src/handlers/deleteCloudFormationStack.ts

```diff
@@ -73,6 +79,25 @@ async function processRecords(records: ParsedRecord<DataItem>[]) {
 
       logger.error(`${err.message} - ${eventDetails}`);
       // throw err;
+
+      // Handles when there is a dependency between some stack deletes
+      // and a successive attempt at deleting will succeed.
+      // So here we recreate the dynamodb record with a shorter TTL
+      // based on the deleteCount
+      if (
+        record.oldData?.deleteCount &&
+        record.oldData.deleteCount > config.MAX_CLEANUP_RETRY
+      ) {
+        // Log message to cloudwatch
+        logger.error(
+          `Failed to delete stack after ${config.MAX_CLEANUP_RETRY} additional attempts: ${oldData.stackName}`
+        );
+      } else {
+        // Recreate record
+        const deleteItem = generateRepeatedDeleteItem(oldData);
```

Comment:
So you adding it back to dynamodb to try again ?

Reply:
Yeah exactly - the TTL will expire/auto delete the record and we need to add it back. Similarly the git webhook deletes the record and then this delete function attempts the CFn delete